drools-parser/src/main/java/org/drools/parser/ParserStringUtils.java

```diff
@@ -31,4 +33,9 @@ public static String getTextPreservingWhitespace(ParserRuleContext ctx) {
         Interval interval = new Interval(startIndex, stopIndex);
         return ctx.start.getTokenSource().getInputStream().getText(interval);
     }
+
+    public static String getTokenTextPreservingWhitespace(ParserRuleContext ctx, TokenStream tokenStream) {
+        // tokenStream is required to get hidden channel token (e.g. whitespace). Unlike getTextPreservingWhitespace, this method reflects Lexer normalizeString
+        return tokenStream.getText(ctx.start, ctx.stop);
+    }
```

Comment:
I need to use `TokenStream` to get normalized string (by `DRLLexer.g4`'s  `normalizeString`).